week2/index.js

```diff
@@ -0,0 +1,163 @@
+"use-strict";
+
+const fs = require('fs');
+const TODO_FILE = process.env['TODO_file'] || 'todo.json'
+const HELP_MENU = 'help.txt';
+
+let readFiles = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve(data);
+
+        })
+    })
+
+};
+
+let writeFiles = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(TODO_FILE, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve();
+        })
+    })
+}
+
+let addTodoItem = (filename, todoList, itemText) => {
+    todoList.push({
+        TASK_NUMBER: todoList.length + 1,

```

Comment:
Why `TASK_NUMBER` and `TODO_TASK`? Surely it would make sense to either have `TODO_NUMBER` and `TODO_TASK`, or `TASK_NUMBER` and `TASK_TODO` or something? The names are inconsistent.

Reply:
Also, I think it's a mistake to assign a `TASK_NUMBER` here. Arrays are ordered lists, so the order is always maintained - that means you can calculate the task number based on the position in the array, instead of having to manage the task numbers yourself (like you do with the for loop on line 61).

I think what you're hinting at is unique IDs, which will get covered in the databases class, so don't worry about it for now.